tools/sql/handler.go

```diff
@@ -104,7 +104,9 @@ func createDatabase(cli *cli.Context, logger log.Logger) error {
 }
 
 func DoCreateDatabase(cfg *config.SQL, name string) error {
-	cfg.DatabaseName = ""
+	if cfg.PluginName != "postgres" {
+		cfg.DatabaseName = ""
```

Comment:
This is the DB used to establish the connection. If the db is what you are about to create, that means it is not exists yet, then how would the connection to that db work?

Reply:
That's actually not completely correct. Yes the postgres db must be running and listening on port 5432 but the db instance does not have to exist at this point. It's actually one of the commands available in the temporal-sql-tool CLI and is the purpose of this method (hence the name DoCreateDatabase). I believe a default instance does have to exist though, which temporal currently assumes is one of two hard-coded values. If you look at how temporal actually connects to the postgres instance, if the db named does exist, it skips the creation, which is great. However, if it doesn't exist yet, it still must connect in order to be able to create the db instance. However, postgres requires you connect to a db instance. In this case, temporal loops through a [hard-coded list of default db instance names](https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/postgresql/plugin.go#L47) to try to connect to one of those to establish a session.

Here's where the looping on default names starts. Given the current code, it will always reach this point since the cfg.DatabaseName is cleared out. In my case, the default db instance for my postgres database is not one of the defaults and I can't change it. 
https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/postgresql/plugin.go#L110

Once a session is established a new db can be created (assuming the credentials have rights to do that).

Reply:
The real confusion I have is that this CLI tool provides a global --database flag I can pass to accommodate the differently named default db I have, but the value I pass for this flag is completely ignored for the create command.

Reply:
That make sense.